tariff-management-service/tariffmanagement-app/src/test/java/com/microservices/tariffmanagement/service/TariffServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+
+        assertEquals(savedTariffDto.getId(), savedTariffEntity.toTariffDto().getId());
+        assertEquals(savedTariffDto.getName(), savedTariffEntity.toTariffDto().getName());
+        assertEquals(savedTariffDto.getPrice(), savedTariffEntity.toTariffDto().getPrice());
+        assertEquals(savedTariffDto.getWashingTime(), savedTariffEntity.toTariffDto().getWashingTime());
+    }
+
+    @Test
+    public void testTariffService_priceIsNegative() throws IllegalArgumentException {
+        CreationTariffDto creationTariffDto = new CreationTariffDto();
+        creationTariffDto.setName(UNIQUE_NAME);
+        creationTariffDto.setPrice(BigDecimal.valueOf(NEGATIVE_PRICE));
+        creationTariffDto.setWashingTime(CORRECT_WASHING_TIME);
+        try {
+            tariffService.createTariff(creationTariffDto);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException thrown) {
+            assertEquals(thrown.getMessage(), EXCEPTION_MESSAGE_FOR_NEGATIVE_VALUES);
+        }
+    }
+
+    @Test
+    public void testTariffService_washingTimeIsNegative() throws IllegalArgumentException {
+        CreationTariffDto creationTariffDto = new CreationTariffDto();
+        creationTariffDto.setName(UNIQUE_NAME);
+        creationTariffDto.setPrice(BigDecimal.valueOf(CORRECT_PRICE));
+        creationTariffDto.setWashingTime(NEGATIVE_WASHING_TIME);
+        try {
+            tariffService.createTariff(creationTariffDto);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException thrown) {
+            assertEquals(thrown.getMessage(), EXCEPTION_MESSAGE_FOR_NEGATIVE_VALUES);
+        }
+    }
+
+    @Test
+    public void testTariffService_priceAndWashingTimeIsNegative() throws IllegalArgumentException {
+        CreationTariffDto creationTariffDto = new CreationTariffDto();
+        creationTariffDto.setName(UNIQUE_NAME);
+        creationTariffDto.setPrice(BigDecimal.valueOf(NEGATIVE_PRICE));
+        creationTariffDto.setWashingTime(NEGATIVE_WASHING_TIME);
+        try {
+            tariffService.createTariff(creationTariffDto);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException thrown) {
+            assertEquals(thrown.getMessage(), EXCEPTION_MESSAGE_FOR_NEGATIVE_VALUES);
+        }
+    }
+
+    @Test
+    public void testTariffService_nameIsNonUnique() throws IllegalArgumentException {
+        CreationTariffDto creationTariffDto = new CreationTariffDto();
+        creationTariffDto.setName(NON_UNIQUE_NAME);
+        creationTariffDto.setPrice(BigDecimal.valueOf(CORRECT_PRICE));
+        creationTariffDto.setWashingTime(CORRECT_WASHING_TIME);
+
+        when(tariffRepository.findByName(NON_UNIQUE_NAME))
+                .thenReturn(Optional.of(new TariffEntity(creationTariffDto)));
+
+        try {
+            tariffService.createTariff(creationTariffDto);
+            fail("Expected IllegalArgumentException");
```

Comment:
What are you trying to achive here? I mean, do you even need fail() here? Just execute creteTariff method (it will throw exception anyway, right?) and then catch it's exception. Also you can use @Expected annotation on test method and do not catch exception at all

Reply:
It's concerns other tests too

Reply:
I'm trying to catch a exception. I wrote this because I thought I would consider all the options. I understood your remark.